stories/util.js

```diff
@@ -62,97 +62,135 @@ Wrap.defaultProps = {
   indexName: 'instant_search',
 };
 
-export const WrapWithHits = ({
-  searchParameters: askedSearchParameters = {},
-  children,
-  searchBox = true,
-  hasPlayground = false,
-  linkedStoryGroup,
-  pagination = true,
-  appId,
-  apiKey,
-  indexName,
-  hitsElement,
-}) => {
-  const sourceCodeUrl = `https://github.com/algolia/react-instantsearch/tree/master/stories/${linkedStoryGroup}.stories.js`;
-  const playgroundLink = hasPlayground ? (
-    <button
-      onClick={linkTo(linkedStoryGroup, 'playground')}
-      className="playground-url"
-    >
-      <span>Play with props</span>
-    </button>
-  ) : null;
+export class WrapWithHits extends PureComponent {
+  static propTypes = {
+    appId: PropTypes.string,
+    apiKey: PropTypes.string,
+    indexName: PropTypes.string,
+    children: PropTypes.node,
+    searchBox: PropTypes.bool,
+    linkedStoryGroup: PropTypes.string,
+    hasPlayground: PropTypes.bool,
+    pagination: PropTypes.bool,
+    searchParameters: PropTypes.object,
+    hitsElement: PropTypes.element,
+    initialSearchState: PropTypes.object,
+    onSearchStateChange: PropTypes.func,
+  };
 
-  const footer = linkedStoryGroup ? (
-    <div className="footer-container">
-      {playgroundLink}
-      <a
-        href={sourceCodeUrl}
-        className="source-code-url"
-        target="_blank"
-        rel="noopener noreferrer"
-      >
-        <div>View source code</div>
-      </a>
-    </div>
-  ) : null;
+  // defaultProps added so that they're displayed in the JSX addon
+  static defaultProps = {
+    appId: 'latency',
+    apiKey: '6be0576ff61c053d5f9a3225e2a90f76',
+    indexName: 'instant_search',
+    pagination: true,
+    searchBox: true,
+    hasPlayground: false,
+    searchParameters: {},
+    initialSearchState: {},
+  };
+
+  constructor(props) {
+    super();
 
-  const hits = hitsElement || <CustomHits />;
+    this.state = {
+      searchState: props.initialSearchState,
+    };
+  }
 
-  const searchParameters = {
-    hitsPerPage: 3,
-    ...askedSearchParameters,
+  onSearchStateChange = ({ configure, ...searchState }) => {
```

Comment:
Why do we filter out `configure`?

Reply:
No reason, I think it's a left-over from a previous implementation, I'll remove it.